src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -356,18 +356,76 @@ protected virtual void AddRecipients(TransactionBuildContext context)
             if (context.Recipients.Any(a => a.Amount == Money.Zero))
                 throw new WalletException("No amount specified.");
 
-            if (context.Recipients.Any(a => a.SubtractFeeFromAmount))
-                throw new NotImplementedException("Substracting the fee from the recipient is not supported yet.");
+            var totalRecipients = context.Recipients.Count(r => r.SubtractFeeFromAmount);
 
-            foreach (Recipient recipient in context.Recipients)
-                context.TransactionBuilder.Send(recipient.ScriptPubKey, recipient.Amount);
+            // If we have any recipients that require a fee to be subtracted from the amount, then
+            // calculate fee and evenly distribute it among all recipients. Any remaining fee should be
+            // subtracted from the first recipient.
+            if (totalRecipients > 0)
+            {
+                Money fee = this.CalculateFee(context);
+                var recipientFee = fee.Satoshi / totalRecipients;
+                var remainingFee = fee.Satoshi % totalRecipients;
+
+                for (int i = 0; i < context.Recipients.Count; i++)
+                {
+                    var recipient = context.Recipients[i];
+
+                    if (recipient.SubtractFeeFromAmount)
+                    {
+                        // First receiver pays the remainder not divisible by output count.
+                        long feeToSubtract = i == 0 ? remainingFee + recipientFee : recipientFee;
+                        long remainingAmount = recipient.Amount.Satoshi - feeToSubtract;
+                        if (remainingAmount <= 0)
+                            throw new WalletException("Fee is higher than amount to send.");
```

Comment:
Can we perhaps just add the amounts into the exception message?

```suggestion
                            throw new WalletException(string.Format("Fee {0} is higher than amount {1} to send.",feeToSubtract, recipient.Amount.Satoshi));
```

Reply:
:+1: